app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/CheckAgencyController.scala

```diff
@@ -154,7 +137,11 @@ class CheckAgencyController @Inject()
         }
       }
     }
-    if(agentAssuranceFlag) processCheckAgencyStatusWithR2DW else processCheckAgencyStatus
+
+    assuranceService.isOnRefusalToDealWithList(knownFacts.utr) flatMap {
```

Comment:
could we make it part of `assuranceService.assureIsAgent`?

Reply:
Yep, and mentioned in the AssuranceResults